### PR TITLE
Sping Course Maximova 

### PR DESCRIPTION
**Pull request by onaxiz**

### CODE REVIEW

File: phonebook.iml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (L-Evg) [inline phonebook.iml]: My eyes ... oh my ... they are bleeding... red-red-red blood drops down to the floor... .

File: src/main/java/com/phonebook/main/InMemoryRepository.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
      * @param name
      * @param phone
      */
-    void addPhone(String name, String phone);
+    void addPhone(String name, String[] phone);
```
Comment (L-Evg) [inline src/main/java/com/phonebook/main/InMemoryRepository.java]: ` void addPhone(String name, String[] phone);`  - please use proper naming. 
 `void addPhone(String name, String phone);`
AND
  `void addPhones(String name, String[] phones);`  (basically I woul use List<String>)
  OR
` void addPhones(String name, String... phones);`

File: src/main/java/com/phonebook/main/PhoneBookMain.java [hunk 3]
```diff
@@ -26,13 +27,37 @@ public static void main(String[] args) {
         renderer.info("type 'exit' to quit.");
         while (sc.hasNext()) {
             String line = sc.nextLine();
-            if (line.equals("exit")) {
-                renderer.info("Have a good day...");
-                break;
-            }
+            System.out.println(line);
```
Comment (L-Evg) [inline src/main/java/com/phonebook/main/PhoneBookMain.java]: Why don't you use the renderer?

File: src/main/java/com/phonebook/main/PhoneBookMain.java [hunk 4]
```diff
@@ -26,13 +27,37 @@ public static void main(String[] args) {
...
-                throw new UnsupportedOperationException("Implement it!");
+                if ("exit".contains(line.toLowerCase())) {
+                    renderer.info("Have a good day...");
+                    break;
+                }
+                if (line.toLowerCase().startsWith("add")) {
+                    String[] consoleArguments = line.split(" ");
+                    String[] phones = consoleArguments[2].split(",");
+                    phoneBook.addNumber(consoleArguments[1], phones);
+                    renderer.info(phoneBook.findAll().toString());
```
Comment (L-Evg) [inline src/main/java/com/phonebook/main/PhoneBookMain.java]: `phoneBook.findAll()` - could return you a long-long list of entries.

File: src/main/java/com/phonebook/main/PhoneBookMain.java [hunk 5]
```diff
@@ -26,13 +27,37 @@ public static void main(String[] args) {
...
+                    break;
+                }
+                if ("show".equals(line.toLowerCase())) {
+                    renderer.info(phoneBook.findAll().entrySet().stream()
+                            .map(e -> e.getKey() + " " + e.getValue())
+                            .collect(Collectors.joining("\n")));
```
Comment (L-Evg) [inline src/main/java/com/phonebook/main/PhoneBookMain.java]: I would add that as a separate method like "prettyPrint"

File: src/main/java/com/phonebook/spring/InMemoryRepositoryIml.java [hunk 6]
```diff
@@ -48,12 +46,18 @@ public String findNameByPhone(String phone) {
     }
 
     @Override
-    public void addPhone(String name, String phone) {
-        throw new UnsupportedOperationException("Implement it!");
+    public void addPhone(String name, String[] phones) {
+        Set<String> phonesSet = new HashSet<String>();
+        for (String phone : phones) {
+            phonesSet.add(phone);
```
Comment (L-Evg) [inline src/main/java/com/phonebook/spring/InMemoryRepositoryIml.java]: Doesn't it have something like `addAll()`?

File: src/main/java/com/phonebook/spring/InMemoryRepositoryIml.java [hunk 7]
```diff
@@ -48,12 +46,18 @@ public String findNameByPhone(String phone) {
...
+        }
+        this.data.put(name, phonesSet);
     }
 
     @Override
-    public void removePhone(String phone) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Implement it!");
+    public void removePhone(String phone) {
+        for (HashMap.Entry<String, Set<String>> pair : this.data.entrySet()){
```
Comment (L-Evg) [inline src/main/java/com/phonebook/spring/InMemoryRepositoryIml.java]: `HashMap.Entry<String, Set<String>> ` - please use interfaces as much as possible.

File: src/main/java/com/phonebook/spring/PhoneBook.java [hunk 8]
```diff
@@ -11,9 +12,10 @@
  * PhoneBook service implementation
  */
 @Service
+@Scope(scopeName = "singleton")
 public class PhoneBook {
 
-    // @Autowired
+//    @Autowired
```
Comment (L-Evg) [inline src/main/java/com/phonebook/spring/PhoneBook.java]: ?

File: src/main/java/com/phonebook/spring/PhoneBook.java [hunk 9]
```diff
@@ -49,4 +51,11 @@ public void setRepository(InMemoryRepository repository) {
     /**
      * TODO: please add required methods here
      */
+    public void deleteNumber(String phone) {
+        repository.removePhone(phone);
+    }
+
+    public void addNumber(String name, String[] phone) {
```
Comment (L-Evg) [inline src/main/java/com/phonebook/spring/PhoneBook.java]: ...Numbers...

**Review by L-Evg — CHANGES_REQUESTED** [review]

**Comment by tbarabanov** [comment]
Заведите свой репозиторий!

**Pull request closed without merging**